### ISSUE TITLE
[Bug] Manual IGDB search has a strange ordering

### ISSUE THREAD

**Issue by helgehatt** [zurdi15/romm#395]
**RomM version**
1.10

**Describe the bug**
The ordering of results from doing a manual IGDB search should be improved. Some of the results are so strange I would actually call it a bug. The first result in the order is also the game selected by the scan. Here are some examples I've experienced:

Search: Portal
Result: Warcraft II Beyond the Dark Portal

Search: Z (yes, the game is just called Z)
Result: Dragon Ball Z: Kakarot

The search should prioritize the shortest full match, such that scanning "Portal 2" becomes "Portal 2" and not "Portal 2: In Motion".

Additionally, I think that games you have manually searched for and selected, should not be overridden by a full scan. If i change multiple filenames on disk, I typically run a full scan, and then I have to manually map all the wrong results over again every time. Maybe a solution would be to store a hash of the underlying files, so that you can keep track of manual game mapping and file renames.

**Comment by GAntoine** [comment]
> The ordering of results from doing a manual IGDB search should be improved

Agreed, the upcoming v2 release will have some improvements to result order, with more to come.

> then I have to manually map all the wrong results over again every time

I don't expect this to happen, unless you're selecting "Complete rescan" when running a scan on a platform. Normal rescan will only write to ROM files it hasn't found.

**Comment by helgehatt** [comment]
> > then I have to manually map all the wrong results over again every time
> 
> I don't expect this to happen, unless you're selecting "Complete rescan" when running a scan on a platform. Normal rescan will only write to ROM files it hasn't found.

Yes, sorry, by full scan I mean a "Complete rescan". It is necessary to do a complete rescan after renaming files on disk, because otherwise you'll end up with a bunch of duplicates, where half of them are pointing to no longer existing files.

**Comment by GAntoine** [comment]
> otherwise you'll end up with a bunch of duplicates

That sounds like something we need to fix, romm should probably remove games that are no longer in the filesystem from the db.

**Comment by helgehatt** [comment]
> romm should probably remove games that are no longer in the filesystem from the db

Maybe just a separate "Clean" button to scan and remove non-existing files?

**Comment by zurdi15** [comment]
Just to let you know, if you rename a file and you do a ``normal`` scan, the no longer exist game entry will be purged automatically and RomM will only show the "new" game. You don't need to do a ``complete rescan`` for that.

Test it and if you achive it I will close the issue.
